### ISSUE TITLE
Where to put the bindkey?

### ISSUE THREAD

**Issue by nbarnwell** [home-assistant/home-assistant.github.io#25695]
### Feedback

Xiaomi LYWSD03MMC temperature/humidity sensor.

I flashed a custom firmware with the Telelink flasher, and obtained the bindkey. I added the device to Home Assistant using the Xiaomi integration but it timed out saying the device hadn't broadcast in 60 seconds, and I'd be prompted for the bindkey if and when it did, but nothing's happened.

Is there some way I can manually add the bindkey, or force it to look again? It seems odd that HA finds the device so quickly, but then can't talk to it straight after.

### URL

https://www.home-assistant.io/integrations/xiaomi_ble/#encryption

**Comment by lg0killer** [comment]
I ran into the same issue.

I eventually got to this link: https://github.com/pvvx/ATC_MiThermometer

After reading and playing around with it, I flashed the Temperature and Humidity sensor with the custom firmware (Mine was v2 but still worked).

Download the firmware on the repo and flash it using this website: https://atc1441.github.io/TelinkFlasher.html
(Just note you need Bluetooth on your PC)

After that, the device won't be recognised by atc1441 anymore, so go to this website to control it: https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html

Then you just need to set the advertising type to BTHome and mark it as encrypted. Save the config and get all the mi keys.

This will then show up on your Home Assistant as BTHome, but you will ultimately have your Temperature, Humidity and power levels of the device

The best thing about the custom firmware is that you can add multiple items on the LCD like battery and clock. Play around with it. Worse case you just need to reset it

**Comment by nbarnwell** [comment]
In the end I worked out a method that didn't require encryption at all. I suppose a suitably motivated attacker could con me into thinking I'm cold when I'm not, but I can live with that for now. I think part of what caught me out was the device name changing to something I didn't recognise upon flashing. Once I realised the flasher was using a certain format for the device name I started to get the hang of it.
